src/main/java/nextstep/subway/applicaion/LineService.java

```diff
@@ -98,7 +91,51 @@ public void deleteLineById(Long id) {
 
     public Line findById(Long id) {
         return lineRepository.findById(id)
-                .orElseThrow(() -> new BadRequestException("존재하지 않는 노선입니다."));
+                .orElseThrow(() -> new BadRequestException("존재하지 않는 노선입니다. lineId = " + id));
+    }
+
+    public SectionResponse addSection(Long lineId, StationsDto stationsDto, int distance) {
+        Line line = findById(lineId);
+        validateAddSection(new Sections(line.getSections()), stationsDto);
+
+        Section section = Section.builder()
+                .line(line)
+                .upStation(stationsDto.getUpStation())
+                .downStation(stationsDto.getDownStation())
+                .distance(distance)
+                .build();
+        line.getSections().add(section);
+
+        return SectionResponse.of(section);
+    }
+
+    private void validateAddSection(Sections sections, StationsDto stationsDto) {
+        if(!sections.isDownStation(stationsDto.getUpStation())) {
+            throw new BadRequestException("새로운 구간의 상행역은 현재 등록되어있는 하행 종점역이어야 합니다.");
+        }
+
+        if(sections.isRegisteredStation(stationsDto.getDownStation())) {
+            throw new BadRequestException("새로운 구간의 하행역은 현재 등록되어있는 역일 수 없습니다.");
+        }
+    }
+
+    public void deleteSection(Long lineId, Station station) {
+        Line line = findById(lineId);
+
+        Sections sections = new Sections(line.getSections());
+        validateDeleteSection(sections, station);
+
+        line.getSections().remove(sections.getLastSection());
```

Comment:
객체 내부의 값을 가져와서 처리하기 보다는 메서드를 호출하여 해결하도록 해보세요.
그렇게 할 경우 아래와 같은 형태로 변경될 것 같습니다.

```suggestion
        line.removeSection(stataion);
```